snc.sh

```diff
@@ -56,6 +56,24 @@ else
     fi
 fi
 
+function retry {
+    local retries=5
```

Comment:
Wondering if there should be a 'max wait time in seconds' parameter rather than hardcoding `5`?

Reply:
@cfergeau yes we can change it when we really required that value, right now with this it is mostly working.